2023-07-03: Add timestamps and answers (#109)
oxidecomputer/oxide-and-friends: 2023_07_03.md

> ([*](https://news.ycombinator.com/item?id=36552561))

A:

> > Bryan: "Elastic infrastructure is great and shouldn't be cloystered to the public cloud";

Reviewer: ```suggestion
> Bryan: "Elastic infrastructure is great and shouldn't be cloistered to the public cloud";
```

> Innovation budget can be a useful concept. And I'm afraid it's being stretched a lot.
> ([*](https://news.ycombinator.com/item?id=36555250))

A:

> > Bryan: We can deliver more value with out own hypervisor;

Reviewer: ```suggestion
> Bryan: We can deliver more value with our own hypervisor; we've had a lot of experience in that domain from Joyent.
```

> [Intel vs. AMD](https://www.theregister.com/2023/01/12/intel_sapphire_rapids/)

> Steve: "We think it's super important that we're very transparent with what we're building"

[@56:05] Q:
> what is the interface I get when I turn this $$$ computer on? What is the zero to first value when I buy this hardware?
> ([*](https://discord.com/channels/1042492311080288306/1125464736893513869/1125590097300049991))

A:

> Steve: "You roll the rack in, you have to give it power, and you have give it networking [...] and you are then off on starting the software experience";
> Large pool of infrastructure reosources for customers/devs/SREs/... in a day or less;
> Similar experience to public cloud providers

[@01:02:06] Q:
> One of my concerns when buying a complete solution like an iPhone (or an Oxide rack 😉) is how much ownership I actually have over the device. Where does Oxide draw the line on "ownership" of the rack?
> ([*](https://discord.com/channels/1042492311080288306/1125464736893513869/1125592711865573426))

A:

> Bryan: "You bought the computer, it belongs to you";
> it is important to us the enable that ownership

> Steve: operator level and developer level;

> > "Anythinh that we are injected ourselves into is [...] in the interest of making sure that the customer is getting a highly available and efficient system"

Reviewer: ```suggestion
> "Anything that we are injected ourselves into is [...] in the interest of making sure that the customer is getting a highly available and efficient system"
```

> Is the rack dependent on any Oxide cloud services?

A:

> Bryan: "Emphatically not"
> The customer has total agency over data I/O

> Adam: Updates will be fetches from a web server

Interlude

[@01:07:49]
> This is super cool. I realize a lot of HN folks might not see the point of this, but it literally saves an entire team of people for companies.
> ([*](https://news.ycombinator.com/item?id=36554036))

[@01:08:00]
> Oxide is such an ambitious project, I am such a fan of the aesthetic and design and of course transparency of all of the cool people that work there.
> ([*](https://news.ycombinator.com/item?id=36552390))

[@01:09:44]
> FYI for language pedants like me: It’s “on-premises” or “on-prem”. A “premise” is something assumed to be true.
> ([*](https://news.ycombinator.com/item?id=36552245))

[![On premises / on premise](https://img.youtube.com/vi/RWHocP0AzMc/default.jpg)](https://youtu.be/RWHocP0AzMc)

[@01:11:43] Q (Chat):
> When is oxide gonna make a smaller rack?

A:

> Bryan: Probably no product for the homelab;
> "Architecturally, there is nothing here that won't scale down';
> ear to the market

> Steve: can be scaled down from a power perspective

> [@01:15:38] Interlude about lack of GPU support.

Reviewer: ```suggestion
[@01:15:38] Bryan discusses how to deliver "Oxide value" with a GPU product.
```

A:

> Bryan: Keeping a close ear to the market;
> "You will see a Genoa-based product"

> Steve: You can add next-gen sleds to the rack, you have modularity

[@01:26:10] Q:
> Do you have any liquid cooling solutions collaborators?
> ([*](https://news.ycombinator.com/item?id=36560169))

A:

> Steve: Collaboration with [Motivair](https://www.motivaircorp.com/products/chilleddoor/);

> > No liquid cooling solution for now

Reviewer: ```suggestion
> The product is conventionally cooled right now.
```

> Looking at what makes sense in future architectures

> Bryan: Rack power budget is a challenge

[@01:28:41] Q:
> Why do you use Illumos?
> ([*](https://news.ycombinator.com/item?id=36552630))

A:

> Bryan: "We need to be able to control our fate";
> [dtrace](https://en.wikipedia.org/wiki/DTrace);

> > "To trust or to mistrust Oracle is to anthropomorphize Larry Ellison"

Reviewer: ```suggestion

Adam: "It was our fastest path to market (both development and support) with our team, expertise, and experience."
```

[@01:41:20] Q:
> Congrats! This seems to be a cool product! How does this work with software updates? Are they subscription-based, or are they for free forever? In case the company goes bankrupt, will the software still work, and is the source code open?
> ([*](https://twitter.com/ISibboI/status/1675058910250643457))

A:

> Bryan: "In many ways we viewed the ability to upgrade as the most important feature"

> > The software is open (for the most part)

Reviewer: ```suggestion
> The software is open by default, and we're reviewing instances where we're constrained by IP we don't own.
```
